pkg/messaging/grpc_proxy.go

```diff
@@ -98,7 +98,8 @@ func (p *proxy) intercept(ctx context.Context, fullName string) (context.Context
 	}
 
 	// proxy to a remote daprd
-	conn, cErr := p.connectionFactory(outCtx, target.address, target.id, target.namespace, false, false, false, grpc.WithDefaultCallOptions(grpc.CallContentSubtype((&codec.Proxy{}).Name())))
+	// connection is recreated because its certification may have already been expired
+	conn, cErr := p.connectionFactory(outCtx, target.address, target.id, target.namespace, false, true, false, grpc.WithDefaultCallOptions(grpc.CallContentSubtype((&codec.Proxy{}).Name())))
```

Comment:
While HTTP invoke recreates connection only after receiving unavailable error, this strategy could not be used for gRPC proxy because it is also used for streaming RPC.

When HTTP invoke receives unavailable error, it can recreate connection and resends the same request to the server as a retry phase.

On the other hand, when gRPC proxy receives unavailable error, it does not know which request should be resended because streaming requests and responses are passed through it asynchronously.

https://github.com/dapr/dapr/blob/a653a35c7bfb2c2b0dca982a8534e674d99b0e3b/pkg/grpc/proxy/handler.go#L82

Reply:
Related: https://github.com/grpc/grpc-go/issues/3021#issuecomment-531337170

Reply:
I'm actually pretty certain your PR will fix the issue. I'll verify this fix next week.

Reply:
@Syuparn the fix works :)